pkg/externalresource/control.go

```diff
@@ -0,0 +1,89 @@
+package externalresource
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/okteto/okteto/pkg/constants"
+	oerrors "github.com/okteto/okteto/pkg/errors"
+	"github.com/okteto/okteto/pkg/externalresource/k8s"
+	"github.com/okteto/okteto/pkg/format"
+	olog "github.com/okteto/okteto/pkg/log"
+	k8sErrors "k8s.io/apimachinery/pkg/api/errors"
+	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// K8sControl represents the controller that performs the actions with k8s
+type K8sControl struct {
+	ClientProvider func(clientcmdapi.Config) (k8s.ExternalResourceV1Interface, error)
+	Cfg            *clientcmdapi.Config
+}
+
+func (c *K8sControl) Deploy(ctx context.Context, name, ns string, externalInfo *ExternalResource) error {
+	k8sclient, err := c.ClientProvider(*c.Cfg)
+	if err != nil {
+		return fmt.Errorf("error creating external CRD client: %s", err.Error())
+	}
+
+	externalResourceCRD := translate(name, externalInfo)
+
+	old, err := k8sclient.ExternalResources(ns).Get(ctx, externalResourceCRD.Name, metav1.GetOptions{})
```

Comment:
Have we tested what happens on a redeploy. Do we really update the existing external resources?